### PR TITLE
Improve the SDS doc.

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

**Comment by myidpt** [comment]
@sdake @rcaballeromx PTAL, thanks!

File: content/docs/tasks/security/auth-sds/index.md [hunk 1]
```diff
@@ -8,20 +8,25 @@ keywords: [security,auth-sds]
...
+
+These issues are addressed in Istio 1.1 through the SDS identity provision flow. The workflow can be described as follows.
+
+The workload sidecar Envoy requests key and certificates from the Citadel Agent (SDS server, which runs as per-node `DaemonSet`) using a Kubernetes
+service account JWT through the SDS API.
+The Citadel Agent generates a key pair and sends the CSR request to Citadel. Citadel verifies the JWT and issues the certificate to the Citadel Agent.
+Eventually, Citadel Agent sends the key and certificate back to the workload sidecar.
+In this approach, the private key never leaves the node; it is only kept in the Citadel Agent and Envoy sidecar's memory.
+Thus, this approach is more secure than the secret volume mount approach, which relies on the secret and local files on pod.
+Also, the sidecar Envoy is able to dynamically renew the key and certificate through the SDS API, so restarting Envoy can be avoided.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]: The list also helps make the actors and actions a lot clearer:
```suggestion
* The sidecar Envoy can dynamically renew the key and certificate through the SDS API: Certificate rotations no longer require Envoy to restart.
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 2]
```diff
@@ -58,12 +63,9 @@ sleep.bar to httpbin.foo: 200
...
 {{< /text >}}
 
-## Verifying no secret-mount file is generated
+## Verifying no secret-volume mounted file is generated
 
-As mentioned, instead of using secret-mount file, key/certificate provisioned by SDS flow are stored in workload sidecar's memory,
-which is more secure and avoids traffic interruption.
-
-Verify no secret-mount file is generated by remotely accessing to workload sidecar container that we just deployed:
+Verify no secret-volume mounted file is generated by remotely accessing to workload sidecar container that we just deployed:
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]: The text became convoluted and difficult to follow with the added adjective. To simplify the language to make the text more understandable I suggest:
```suggestion
To verify that no secret-volume mounted file is generated, remotely access the deployed workload sidecar container:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 3]
```diff
@@ -80,3 +82,11 @@ $ kubectl delete ns foo
 $ kubectl delete ns bar
 $ kubectl delete -f istio-auth-sds.yaml
 {{< /text >}}
+
+## Caveats
+
+Currently, the SDS identity provision flow has the following caveats:
+
+* The control plane security has to be turned off. Enabling SDS with control plane security turned on is working in progress.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]: The text fails to communicate how the one impacts the other. Make that impact explicit for clarity:
```suggestion
* To enable SDS, you must turn off the control plane security. Enabling SDS while the control plane security is on remains a work in progress.
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 4]
```diff
@@ -80,3 +82,11 @@ $ kubectl delete ns foo
...
+
+* The support clusters smoothly migrating from using the secret volume mount to using the SDS is working in progress.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]: What's the caveat here? I'm guessing that the migration is not smooth but the text does not clearly convey that idea. 
```suggestion
* There is no smooth migration for support clusters to stop using the secret volume mount approach. A smooth migration for the support clusters to use the SDS approach is a work in progress.
```

**Pull request merged**